### PR TITLE
BEG-80: add subscriber log cleanup cron

### PR DESCRIPTION
**Pull request by anthony-dinino-aligent**
### Description
- We need a cron job that runs periodically and deletes old records from the async_event_subscriber_log table. 
- The setting of how old should be configurable via the admin (older than 10 days, 20 days, 30 days etc).
- For now we can say the cron runs once per day.
- The cron should also be able to be turned on or off.
### Screenshots
Recording 2022-06-08 at 14 11 24
### Notes to reviewers
Jira: https://aligent.atlassian.net/browse/BEG-80
### Time tracking
Please use `BEG-80: magento-async-events Code Review`

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -5,12 +5,25 @@
...
             <resource>Aligent_AsyncEvents::manage</resource>
-            <group id="async_events" translate="label" type="text" sortOrder="1000" showInDefault="1">
+            <group id="async_events" translate="label" type="text" sortOrder="1000" showInDefault="1" showInStore="1" showInWebsite="1">
                 <label>Async Events</label>
-                <field id="max_deaths" translate="label" type="text" sortOrder="10" showInDefault="1">
+                <field id="max_deaths" translate="label" type="text" sortOrder="10" showInDefault="1" showInStore="1" showInWebsite="1">
                     <label>Maximum Deaths</label>
                     <validate>integer validate-not-negative-number</validate>
                 </field>
+                <field id="subscriber_log_cleanup_cron" translate="label" type="select" sortOrder="20" showInDefault="1" showInStore="1" showInWebsite="1">
```
Comment (gowrizrh) [inline etc/adminhtml/system.xml]: These configuration values can just be at the global level instead of website and store views.

File: etc/adminhtml/system.xml [hunk 2]
```diff
@@ -5,12 +5,25 @@
...
+                    <label>Subscriber Log Cleanup</label>
+                    <comment>Enable or disable the subscriber log cleanup cron. </comment>
+                    <source_model>Magento\Config\Model\Config\Source\Enabledisable</source_model>
+                </field>
+                <field id="subscriber_log_cron_delete_period" translate="label" type="text" sortOrder="25" showInDefault="1" showInStore="1" showInWebsite="1">
```
Comment (gowrizrh) [inline etc/adminhtml/system.xml]: same :arrow_up:

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -5,12 +5,25 @@
         <section id="system" translate="label" type="text" sortOrder="900" showInDefault="1">
             <tab>advanced</tab>
             <resource>Aligent_AsyncEvents::manage</resource>
-            <group id="async_events" translate="label" type="text" sortOrder="1000" showInDefault="1">
+            <group id="async_events" translate="label" type="text" sortOrder="1000" showInDefault="1" showInStore="1" showInWebsite="1">
                 <label>Async Events</label>
-                <field id="max_deaths" translate="label" type="text" sortOrder="10" showInDefault="1">
+                <field id="max_deaths" translate="label" type="text" sortOrder="10" showInDefault="1" showInStore="1" showInWebsite="1">
```
Comment (gowrizrh) [inline etc/adminhtml/system.xml]: same :arrow_down:

File: etc/adminhtml/system.xml [hunk 4]
```diff
@@ -5,12 +5,25 @@
...
+                <field id="subscriber_log_cleanup_cron" translate="label" type="select" sortOrder="20" showInDefault="1" showInStore="1" showInWebsite="1">
+                    <label>Subscriber Log Cleanup</label>
+                    <comment>Enable or disable the subscriber log cleanup cron. </comment>
+                    <source_model>Magento\Config\Model\Config\Source\Enabledisable</source_model>
+                </field>
+                <field id="subscriber_log_cron_delete_period" translate="label" type="text" sortOrder="25" showInDefault="1" showInStore="1" showInWebsite="1">
+                    <depends>
+                        <field id="system/async_events/subscriber_log_cleanup_cron">1</field>
+                    </depends>
+                    <label>Oldest Subscriber Log (days)</label>
```
Comment (gowrizrh) [inline etc/adminhtml/system.xml]: Maybe reword this to something like Asynchronous Event Log Lifetime (Days) or something similar.

File: etc/adminhtml/system.xml [hunk 5]
```diff
@@ -5,12 +5,25 @@
...
+                    <comment>Specify how old subscriber logs can get in days. Any logs older will be deleted by the cleanup cron.</comment>
```
Comment (gowrizrh) [inline etc/adminhtml/system.xml]: If you reword the label, you can remove the first sentence.

**Review by gowrizrh — COMMENTED** [review]
Thanks for working on this! I've had a simple look through and everything looks good to me. I might pull the changes locally and test it.

**Comment by anthony-dinino-aligent** [comment]
@gowrizrh Just saw your comments now :) made the changes. Have a look

**Review by gowrizrh — APPROVED** [review]
Sorry for putting this off for so long. Looks good to me :+1:
